src/Network/Network/NetworkWatcher/ConnectionMonitor/ConnectionMonitorBaseCmdlet.cs

```diff
@@ -88,5 +109,637 @@ public MNM.NetworkWatcher GetNetworkWatcherByLocation(string location)
+        }
+
+        public PSConnectionMonitorResultV1 MapConnectionMonitorResultToPSConnectionMonitorResultV1(ConnectionMonitorResult ConnectionMonitor)
+        {
+            PSConnectionMonitorResultV1 psConnectionMonitor = new PSConnectionMonitorResultV1()
+            {
+                Name = ConnectionMonitor.Name,
+                Id = ConnectionMonitor.Id,
+                Etag = ConnectionMonitor.Etag,
+                ProvisioningState = ConnectionMonitor.ProvisioningState,
+                Type = ConnectionMonitor.Type,
+                Location = ConnectionMonitor.Location,
+                AutoStart = ConnectionMonitor.AutoStart,
+                MonitoringIntervalInSeconds = ConnectionMonitor.MonitoringIntervalInSeconds,
+                StartTime = ConnectionMonitor.StartTime,
+                MonitoringStatus = ConnectionMonitor.MonitoringStatus,
+                Tags = NetworkResourceManagerProfile.Mapper.Map<Dictionary<string, string>>(ConnectionMonitor.Tags),
+                ConnectionMonitorType = ConnectionMonitor.ConnectionMonitorType,
+                Source = NetworkResourceManagerProfile.Mapper.Map<PSConnectionMonitorSource>(ConnectionMonitor.Source),
+                Destination = NetworkResourceManagerProfile.Mapper.Map<PSConnectionMonitorDestination>(ConnectionMonitor.Destination),
+            };
+
+            return psConnectionMonitor;
+        }
+
+        public MNM.ConnectionMonitorTestConfiguration GetTestConfigurationByName(IList<ConnectionMonitorTestConfiguration> TestConfigurations, string TestConfigName)
+        {
+            if (TestConfigurations == null || string.IsNullOrEmpty(TestConfigName))
+                return null;
+
+            foreach (ConnectionMonitorTestConfiguration TestConfiguration in TestConfigurations)
+            {
+                if (string.Compare(TestConfigName, TestConfiguration.Name) == 0)
+                {
+                    return TestConfiguration;
+                }
+            }
+
+            return null;
+        }
+
+        public MNM.ConnectionMonitorEndpoint GetEndpoinByName(IList<ConnectionMonitorEndpoint> Endpoints, string EndpointName)
+        {
+            if (Endpoints == null || string.IsNullOrEmpty(EndpointName))
+                return null;
+
+            foreach (ConnectionMonitorEndpoint Endpoint in Endpoints)
+            {
+                if (string.Compare(EndpointName, Endpoint.Name) == 0)
+                {
+                    return Endpoint;
+                }
+            }
+
+            return null;
+        }
+
+    public bool ValidateConnectionMonitorV1V2Parameters(string SourceResourceId, string DestinationResourceId, PSConnectionMonitorResult InputObject, string DestinationAddress, int? MonitoringIntervalInSeconds, List<PSNetworkWatcherConnectionMonitorTestGroupObject> TestGroups, List<PSNetworkWatcherConnectionMonitorOutputObject> Outputs)
+        {
+            if ((!string.IsNullOrEmpty(SourceResourceId)     || 
```

Comment:
Can this be defined using parameter sets?

Reply:
The idea is here is not to validate a single parameter but I test if it is V1 or V2 and for each validate all its parameter sets.

Reply:
redone this